nixos/modules/config/users-groups.nix

```diff
@@ -473,14 +475,15 @@ in {
       '';
     };
 
-    # FIXME: obsolete - will remove.
-    security.initialRootPassword = mkOption {
-      type = types.str;
-      default = "!";
-      example = "";
-      visible = false;
-    };
 
+    users.allowLoginless = mkOption {
```

Comment:
I'm not fond of this option name since I don't know what "loginless" means.

Reply:
Maybe `allowNoPasswordLogin` would be better?

Reply:
What about ssh keys though? Loginless is more accurate, even if not obvious.

Reply:
I mean, I can't really parse "loginless". Maybe invert the meaning and rename to `requireLoginUser`?

Reply:
I also have a general problem with the usage of the term login here. For example the command to get a root shell in a nixos container without any authentication inside the container is called `nixos-container root-login`. You are logging in, you are just not authenticating. Maybe the option should explicitly mention password authentication since this is what it's really about? 
Would `requirePasswordedUser` be alright?

Reply:
> Would requirePasswordedUser be alright?

An ssh key is also sufficient, so that name is not accurate.

The container profile module can disable this check. I suppose it could be refactored so that all ways to satisfy the check are written in the same way, open for extension, but I don't want this idea to hinder progress.